src/app.rs

```diff
@@ -71,6 +82,8 @@ impl App {
                 .map(|prog| (prog.id.clone(), prog))
                 .collect();
 
+            let filter = filter.lock().unwrap().value().to_lowercase();
```

Comment:
I believe by re-assigning filter you are immediately dropping the lock, which is what we want. But I would prefer doing this more explicitly, something like:
```
let filter = filter.lock.unwrap();
let filter_str = filter.value().to_lowercase():
drop(filter);

Reply:
The explicit mutex drop is added now. 👍